File: app/src/main/java/de/cineaste/android/viewholder/series/EpisodeViewHolder.java
package de.cineaste.android.viewholder.series;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageButton;
import android.widget.TextView;

import de.cineaste.android.R;
import de.cineaste.android.entity.series.Episode;

public class EpisodeViewHolder extends RecyclerView.ViewHolder {

    private TextView episodeNumber;
    private TextView episodeTitle;
    private TextView description;
    private ImageButton showDescription;
    private ImageButton hideDescription;
    private CheckBox checkBox;
    private OnEpisodeWatchStateChangeListener onEpisodeWatchStateChangeListener;
    private OnDescriptionShowToggleListener onDescriptionShowToggleListener;

    public interface OnEpisodeWatchStateChangeListener {
        void watchStateChanged(Episode episode);
    }

    public interface OnDescriptionShowToggleListener {
        void showDescription(ImageButton showDescription, ImageButton hideDescription, TextView description);
        void hideDescription(ImageButton showDescription, ImageButton hideDescription, TextView description);
    }

    public EpisodeViewHolder(View itemView, OnEpisodeWatchStateChangeListener onEpisodeWatchStateChangeListener, OnDescriptionShowToggleListener onDescriptionShowToggleListener) {
        super(itemView);

        this.onEpisodeWatchStateChangeListener = onEpisodeWatchStateChangeListener;
        this.onDescriptionShowToggleListener = onDescriptionShowToggleListener;
        episodeNumber = itemView.findViewById(R.id.episodeNumber);
        episodeTitle = itemView.findViewById(R.id.episodeTitle);
        description = itemView.findViewById(R.id.description);
        showDescription = itemView.findViewById(R.id.show_more);
        hideDescription = itemView.findViewById(R.id.show_less);
        checkBox = itemView.findViewById(R.id.checkBox);
    }

    public void assignData(final Episode episode) {
        episodeNumber.setText(String.valueOf(episode.getEpisodeNumber()));
        episodeTitle.setText(episode.getName());
        description.setText(episode.getDescription());
        checkBox.setChecked(episode.isWatched());

        showDescription.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onDescriptionShowToggleListener.showDescription(showDescription, hideDescription, description);
            }
        });

        hideDescription.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                android.util.Log.d("mgr", "hide");
                onDescriptionShowToggleListener.hideDescription(showDescription, hideDescription, description);
            }
        });

        checkBox.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onEpisodeWatchStateChangeListener.watchStateChanged(episode);
            }
        });
    }
}

File: app/src/main/kotlin/de/cineaste/android/viewholder/series/EpisodeViewHolder.kt
package de.cineaste.android.viewholder.series

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.CheckBox
import android.widget.ImageButton
import android.widget.TextView

import de.cineaste.android.R
import de.cineaste.android.entity.series.Episode

class EpisodeViewHolder(
    itemView: View,
    private val onEpisodeWatchStateChangeListener: OnEpisodeWatchStateChangeListener,
    private val onDescriptionShowToggleListener: OnDescriptionShowToggleListener,
    private val context: Context
) : RecyclerView.ViewHolder(itemView) {

    private val episodeTitle: TextView = itemView.findViewById(R.id.episodeTitle)
    private val description: TextView = itemView.findViewById(R.id.description)
    private val showDescription: ImageButton = itemView.findViewById(R.id.show_more)
    private val hideDescription: ImageButton = itemView.findViewById(R.id.show_less)
    private val checkBox: CheckBox = itemView.findViewById(R.id.checkBox)

    interface OnEpisodeWatchStateChangeListener {
        fun watchStateChanged(episode: Episode)
    }

    interface OnDescriptionShowToggleListener {
        fun showDescription(
            showDescription: ImageButton,
            hideDescription: ImageButton,
            description: TextView
        )

        fun hideDescription(
            showDescription: ImageButton,
            hideDescription: ImageButton,
            description: TextView
        )
    }

    fun assignData(episode: Episode) {
        episodeTitle.text = episode.name
        checkBox.isChecked = episode.isWatched

        if (episode.description.isNullOrBlank()) {
            description.text = context.getString(R.string.noDescription)
        } else {
            description.text = episode.description
        }

        showDescription.setOnClickListener {
            onDescriptionShowToggleListener.showDescription(
                showDescription,
                hideDescription,
                description
            )
        }

        hideDescription.setOnClickListener {
            onDescriptionShowToggleListener.hideDescription(
                showDescription,
                hideDescription,
                description
            )
        }

        checkBox.setOnClickListener { onEpisodeWatchStateChangeListener.watchStateChanged(episode) }
    }
}
